src/SelfServe/SqlX/SqlX.rp.ts

```diff
@@ -1,33 +1,98 @@
-  throw new Error("getRegionSpecificMinInstances not implemented");
+export const getPath = (subscriptionId: string, resourceGroup: string, name: string): string => {
+  return `/subscriptions/${subscriptionId}/resourceGroups/${resourceGroup}/providers/Microsoft.DocumentDB/databaseAccounts/${name}/services/sqlx`;
+};
+
+export const updateDedicatedGatewayResource = async (sku: string, instances: number): Promise<string> => {
+  const path = getPath(userContext.subscriptionId, userContext.resourceGroup, userContext.databaseAccount.name);
+  const body: UpdateDedicatedGatewayRequestParameters = {
+    properties: {
+      instanceSize: sku,
+      instanceCount: instances,
+      serviceType: "Sqlx",
+    },
+  };
+  const armRequestResult = await armRequestWithoutPolling({
+    host: configContext.ARM_ENDPOINT,
+    path,
+    method: "PUT",
+    apiVersion,
+    body,
+  });
+  return armRequestResult.operationStatusUrl;
 };
 
-export const getRegionSpecificMaxInstances = async (): Promise<number> => {
-  // TODO: write RP call to get max number of instances needed for this region
-  throw new Error("getRegionSpecificMaxInstances not implemented");
+export const deleteDedicatedGatewayResource = async (): Promise<string> => {
+  const path = getPath(userContext.subscriptionId, userContext.resourceGroup, userContext.databaseAccount.name);
+  const armRequestResult = await armRequestWithoutPolling({
+    host: configContext.ARM_ENDPOINT,
+    path,
+    method: "DELETE",
+    apiVersion,
+  });
+  return armRequestResult.operationStatusUrl;
 };
 
-export const updateDedicatedGatewayProvisioning = async (sku: string, instances: number): Promise<void> => {
-  // TODO: write RP call to update dedicated gateway provisioning
-  throw new Error(
-    `updateDedicatedGatewayProvisioning not implemented. Parameters- sku: ${sku}, instances:${instances}`
-  );
+export const getDedicatedGatewayResource = async (): Promise<SqlxServiceResource> => {
+  const path = getPath(userContext.subscriptionId, userContext.resourceGroup, userContext.databaseAccount.name);
+  const armRequestResult = await armRequestWithoutPolling<SqlxServiceResource>({
+    host: configContext.ARM_ENDPOINT,
+    path,
+    method: "GET",
+    apiVersion,
+  });
+  return armRequestResult.result;
 };
 
-export const initializeDedicatedGatewayProvisioning = async (): Promise<DedicatedGatewayResponse> => {
-  // TODO: write RP call to initialize UI for dedicated gateway provisioning
-  throw new Error("initializeDedicatedGatewayProvisioning not implemented");
+export const getCurrentProvisioningState = async (): Promise<DedicatedGatewayResponse> => {
+  try {
+    const response = await getDedicatedGatewayResource();
+    return {
+      sku: response.properties.instanceSize,
+      instances: response.properties.instanceCount,
+      status: response.properties.status,
+      endpoint: response.properties.sqlxEndPoint,
+    };
+  } catch (e) {
```

Comment:
>catch [](start = 4, length = 5)

@srinathnarayanan : I believe you have an open ADO item to track logging.

Reply:
yes, will have a separate PR for telemetry.